### PR TITLE
post_test: use testify

### PR DESCRIPTION
**Pull request by sebastienD**
#### Summary
Update ```model/post_test.go``` to use ```testify```.

#### Ticket Link
Fixes [#12571](https://github.com/mattermost/mattermost-server/issues/12571)
[MM-19137](https://mattermost.atlassian.net/browse/MM-19137)

### CODE REVIEW

File: model/post_test.go [hunk 1]
```diff
@@ -121,15 +105,11 @@ func TestPostIsSystemMessage(t *testing.T) {
 	post1 := Post{Message: "test_1"}
 	post1.PreSave()
 
-	if post1.IsSystemMessage() {
-		t.Fatalf("TestPostIsSystemMessage failed, expected post1.IsSystemMessage() to be false")
-	}
+	require.False(t, post1.IsSystemMessage(), "post1.IsSystemMessage() should be false")
```
Comment (mgdelacroix) [inline model/post_test.go]: nit: as the check already states that the value should be false, the message string is redundant and can be removed

File: model/post_test.go [hunk 2]
```diff
@@ -121,15 +105,11 @@ func TestPostIsSystemMessage(t *testing.T) {
...
 
 	post2 := Post{Message: "test_2", Type: POST_JOIN_LEAVE}
 	post2.PreSave()
-	if !post2.IsSystemMessage() {
-		t.Fatalf("TestPostIsSystemMessage failed, expected post2.IsSystemMessage() to be true")
-	}
+	require.True(t, post2.IsSystemMessage(), "post2.IsSystemMessage() should be true")
```
Comment (mgdelacroix) [inline model/post_test.go]: same here

File: model/post_test.go [hunk 3]
```diff
@@ -144,9 +124,7 @@ func TestPostSanitizeProps(t *testing.T) {
 
 	post1.SanitizeProps()
 
-	if post1.Props[PROPS_ADD_CHANNEL_MEMBER] != nil {
-		t.Fatal("should be nil")
-	}
+	require.Nil(t, post1.Props[PROPS_ADD_CHANNEL_MEMBER], "should be nil")
```
Comment (mgdelacroix) [inline model/post_test.go]: same here

File: model/post_test.go [hunk 4]
```diff
@@ -157,9 +135,7 @@ func TestPostSanitizeProps(t *testing.T) {
 
 	post2.SanitizeProps()
 
-	if post2.Props[PROPS_ADD_CHANNEL_MEMBER] != nil {
-		t.Fatal("should be nil")
-	}
+	require.Nil(t, post2.Props[PROPS_ADD_CHANNEL_MEMBER], "should be nil")
```
Comment (mgdelacroix) [inline model/post_test.go]: same here

File: model/post_test.go [hunk 5]
```diff
@@ -171,13 +147,8 @@ func TestPostSanitizeProps(t *testing.T) {
...
 
-	if post3.Props[PROPS_ADD_CHANNEL_MEMBER] != nil {
-		t.Fatal("should be nil")
-	}
-
-	if post3.Props["attachments"] == nil {
-		t.Fatal("should not be nil")
-	}
+	require.Nil(t, post3.Props[PROPS_ADD_CHANNEL_MEMBER], "should be nil")
+	require.NotNil(t, post3.Props["attachments"], "should not be nil")
```
Comment (mgdelacroix) [inline model/post_test.go]: same in these two lines

**Comment by jasonblais** [comment]
Hey @sebastienD, let us know if you have any questions about the feedback above. We'd be happy to help! :)

**Comment by jasonblais** [comment]
Hey @sebastienD, we haven't heard back for a few weeks so we'll assume this no longer being worked. We'll close the PR for now.

If you're still interested working on it, let us know in the GitHub issue! :)

**Pull request closed without merging**